### DISCUSSION TITLE
What happens when an x509 CA certificate expires?

### DISCUSSION THREAD

**Discussion by duxiaocheng** [openssl/openssl#23305] [Q&A]
Hello, all,

I know that server's expired certificate would result in TLS handshake failure. 
How about the local trusted CA certificate expires? Does it also cause the TLS handshake failure?
For the best practice, do we need to check the local CA's expiration periodically？

Thanks！
Cheng

**Comment by mattcaswell** [comment]
Certificates are validated at the point that the handshake occurs. This includes all CA certificates. If a CA certificate expires then any subsequent handshakes where the certificate uses that CA in the chain will fail. Handshakes that are already completed are not affected (unless a renegotiation occurs).
